cocotb/share/lib/gpi_log/gpi_logging.c

```diff
@@ -159,14 +159,18 @@ void gpi_log(const char *name, long level, const char *pathname, const char *fun
         PyGILState_Release(gstate);
         return;
     }
-
-    if (filter_ret == Py_True) {
-        Py_DECREF(filter_ret);
+    int is_enabled = PyObject_IsTrue(filter_ret);
+    Py_DECREF(filter_ret);
+    if (is_enabled < 0) {
+        PyErr_Print();
         PyGILState_Release(gstate);
         return;
     }
 
-    Py_DECREF(filter_ret);
+    if (!is_enabled) {
+        PyGILState_Release(gstate);
+        return;
+    }
```

Comment:
That code is slightly confusing; I didn't expect `PyObject_IsTrue()` to return -1 in my initial read of the code, and then it didn't make much sense. How about making this a bit clearer by
- checking for `is_enabled == -1` for the error condition, and add a comment like "unable to determine if |filter_ret| is true or false"
- checking for `is_enabled == 0` for the false case.

Reply:
Thanks for adding a comment. People reading this file are not typically reading cpython source code, so a comment can help reduce confusion here.